delta-app/src/main/java/io/cdap/delta/app/DeltaPipelineId.java

```diff
@@ -22,9 +22,9 @@
 public class DeltaPipelineId {
   private final String namespace;
   private final String app;
-  private final String generation;
+  private final long generation;
 
-  public DeltaPipelineId(String namespace, String app, String generation) {
+  public DeltaPipelineId(String namespace, String app, long generation) {
```

Comment:
One question, why we change the data type to long here?

Reply:
I'll add some comments on how it's being used, but basically we don't have a hook in CDAP that lets an application run some logic when it is deleted, which means we don't have a way to delete the state (offsets, etc). If a replicator is deleted, then re-created with the same name, we don't want to start from the old offset, so we generate this number and use it in the file path for the state. 

In order to figure out which is the current state, the generation needs to be monotonically increasing instead of a random string, which is why it's being changed here.